File: csharp/DVSClient/DVSClient/address/lookup/LookupResult.cs
using DVSClient.Common;
using DVSClient.Server.Address.Lookup;

namespace DVSClient.Address.Lookup
{
    public class LookupResult
    {
        public bool MoreResultsAvailable { get; }
        public AddressConfidence? Confidence { get; }
        public string? SuggestionsKey { get; }
        public string? SuggestionsPrompt { get; }
        public IEnumerable<LookupSuggestion>? Suggestions { get; }
        public IEnumerable<LookupAddressSuggestionV2>? Addresses { get; }
        public IEnumerable<LookupV2ResultAddressFormatted>? AddressesFormatted { get; }
        public string? ReferenceId { get; }

        public LookupResult(RestApiAddressLookupV2Response response) 
        {
            var apiResult = response.Result;
            if (apiResult != null)
            {
                MoreResultsAvailable = apiResult.MoreResultsAvailable;
                Confidence = apiResult.Confidence?.GetEnumValueFromJsonName<AddressConfidence>();
                SuggestionsKey = apiResult.SuggestionsKey;
                SuggestionsPrompt = apiResult.SuggestionsPrompt;
                Suggestions = apiResult.Suggestions?.Select(s => new LookupSuggestion(s)).ToList();
                Addresses = apiResult.Addresses?.Select(a => new LookupAddressSuggestionV2(a)).ToList();
                AddressesFormatted = apiResult.AddressesFormatted?.Select(a => new LookupV2ResultAddressFormatted(a)).ToList();
            }
            else
            {
                MoreResultsAvailable = false;
                Confidence = null;
                SuggestionsKey = string.Empty;
                SuggestionsPrompt = string.Empty;
                Suggestions = new List<LookupSuggestion>();
                Addresses = new List<LookupAddressSuggestionV2>();
                AddressesFormatted = new List<LookupV2ResultAddressFormatted>();
            }

            ReferenceId = response.ReferenceId;
        }
    }
}

File: typescript/src/address/lookup/lookupResult.ts

import { RestApiAddressLookupV2Response } from "../../server/address/lookup/restApiAddressLookupV2Response"
import { AddressConfidence, lookupConfidence } from "../addressConfidence";
import { LookupAddressSuggestionV2, restApiResponseToLookupAddressSuggestionV2Hub } from "./lookupAddressSuggestionV2";
import { LookupSuggestion, restApiResponseToLookupSuggestion } from "./lookupSuggestion";
import { LookupV2ResultAddressFormatted, restApiResponseToLookupAddressFormatted } from "./lookupV2ResultAddressFormatted"

export type LookupResult = {
    moreResultsAvailable: boolean,
    confidence?: AddressConfidence,
    suggestionsKey?: string,
    suggestionsPrompt?: string,
    suggestions?: LookupSuggestion[],
    addresses?: LookupAddressSuggestionV2[],
    addressesFormatted?: LookupV2ResultAddressFormatted[]
}

export function restApiResponseToLookupResult(response: RestApiAddressLookupV2Response): LookupResult {

    const apiResult = response.result;
    if (apiResult) {

        const result: LookupResult = {
            moreResultsAvailable: apiResult.more_results_available
        }
        if (apiResult.confidence) {
            result.confidence = lookupConfidence(apiResult.confidence);
        }
        if (apiResult.suggestions_key) {
            result.suggestionsKey = apiResult.suggestions_key;
        }
        if (apiResult.suggestions_prompt) {
            result.suggestionsPrompt = apiResult.suggestions_prompt;
        }
        if (apiResult.suggestions) {
            result.suggestions = apiResult.suggestions.map(s => restApiResponseToLookupSuggestion(s));
        }
        if (apiResult.addresses) {
            result.addresses = apiResult.addresses.map(a => restApiResponseToLookupAddressSuggestionV2Hub(a));
        }
        if (apiResult.addresses_formatted) {
            result.addressesFormatted = apiResult.addresses_formatted.map( af => restApiResponseToLookupAddressFormatted(af));
        }
        return result;
    } else {
        return {moreResultsAvailable: false}
    }

}